menpo3d/morphablemodel/fitter.py

```diff
@@ -0,0 +1,382 @@
+            else: 
+                dp_dgamma = dp_dalpha
+            
+            if n_betas > 0 and n_alphas > 0:
+                dt = -dt_dbeta
+                SD_gamma = compute_sd(dp_dgamma, VI_dx_uv, VI_dy_uv)
+                SD_img = np.hstack((SD_gamma, dt))
+            elif n_alphas > 0 and n_betas == 0:
+                SD_img = compute_sd(dp_dgamma, VI_dx_uv, VI_dy_uv)
+            else:
+                SD_img = -dt_dbeta              
+                
+            # Hessian approximation
+            H_img = compute_hessian(SD_img)
+
+            # Compute error
+            img_error_uv = img_uv - tex_uv.T
+
+            # Compute steepest descent matrix
+            SD_error_img = compute_sd_error(SD_img, img_error_uv)
+            
+            # Compute and store the error for future plots
+            eps = (img_error_uv ** 2).mean()
+            errors.append(eps)
+
+            # Prior probabilities over shape parameters
+            if camera_update:
+                prior_error = np.concatenate((alpha, rho_array, beta))
+            else:
+                prior_error = np.concatenate((alpha, beta))
+                
+            # Prior probability SD matrices
+            SD_error_alpha_prior = SD_alpha_prior*prior_error
+            SD_error_beta_prior = SD_beta_prior*prior_error
+
+            # Final hessian and SD error matrix
+            H = H_img + 1e-2*H_alpha_prior + H_beta_prior
+            SD_error = SD_error_img + 1e-2*SD_error_alpha_prior + SD_error_beta_prior
+
+            # Compute increment
+            delta_sigma = -np.dot(np.linalg.inv(H), SD_error)
+
+            # Update parameters
+            if camera_update:
+                alpha += delta_sigma[:n_alphas]
+                rho_array += delta_sigma[n_alphas:n_alphas + len(rho_array)]
+                beta += delta_sigma[n_alphas + len(rho_array):n_alphas + len(rho_array) + n_betas]
+            else:
+                alpha += delta_sigma[:n_alphas]
+                beta += delta_sigma[n_alphas:]      
+
+            # Generate the updated instance
+            # The texture is scaled by 255 to cancel the 1./255 scaling in the model class
+            instance = self._model.instance(alpha=alpha, beta=255.*beta)
+            
+            # Clip to avoid out of range pixels
+            instance.colours = np.clip(instance.colours, 0, 1)
+
+            if camera_update:
+                # Compute new view matrix
+                _, R = compute_view_matrix(rho_array)
+                view_t.h_matrix[1:3, :3] = -R.h_matrix[1:3, :3]
+                view_t.h_matrix[0, :3] = R.h_matrix[0, :3]
+            
+                # Update the rasterizer
+                rasterizer.set_view_matrix(view_t.h_matrix)
+            
+            k += 1
+            
+        # Rasterize the final mesh
+        rasterized_result = rasterizer.rasterize_mesh(instance)
+        
+        # Save final values
+        self.rasterized_result = rasterized_result
```

Comment:
don't save the final results to `self`, return them from this method. If there are a number of results, use a dictionary as [in this example](https://github.com/menpo/menpo3d/blob/master/menpo3d/correspond/nicp.py#L246).